### PR TITLE
Add CMake and some instructions.

### PR DESCRIPTION
**Pull request by Odyssey346**
Doesn't work properly on Windows yet (cope with WSL2 for now)

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,10 +1,55 @@
...
+*.o
+*.obj
 
+# Precompiled Headers
+*.gch
+*.pch
 
-output.txt
-error.txt
-.gitignore
```
Comment (Midou36O) [inline .gitignore:10]: Why are these removed?

Comment (Odyssey346) [reply .gitignore:10]: who the hell ignores .gitignore bruh

Comment (Odyssey346) [reply .gitignore:10]: image
also

Comment (Midou36O) [reply .gitignore:10]: Alright, idk why did .gitignore get ignored for some reason so :shrug:
